freqtrade/exchange/hyperliquid.py

```diff
@@ -57,12 +57,83 @@ def _ccxt_config(self) -> dict:
         config.update(super()._ccxt_config)
         return config
 
+    def _get_configured_hip3_dexes(self) -> list[str]:
+        """Get list of configured HIP-3 DEXes."""
+        return self._config.get("exchange", {}).get("hip3_dexes", [])
+
+    def validate_config(self, config: dict) -> None:
+        """Validate HIP-3 configuration at bot startup."""
+        super().validate_config(config)
+
+        configured = self._get_configured_hip3_dexes()
+        if not configured or not self.markets:
+            return
+
+        available = {
+            m.get("info", {}).get("dex")
+            for m in self.markets.values()
+            if m.get("info", {}).get("hip3")
+        }
+        available.discard(None)
+
+        invalid = set(configured) - available
+        if invalid:
+            raise OperationalException(
+                f"Invalid HIP-3 DEXes configured: {sorted(invalid)}. "
+                f"Available DEXes: {sorted(available)}. "
+                f"Check your 'hip3_dexes' configuration!"
+            )
+
     def market_is_tradable(self, market: dict[str, Any]) -> bool:
-        parent_check = super().market_is_tradable(market)
+        """Check if market is tradable, including HIP-3 markets."""
+        super().market_is_tradable(market)
+
+        market_info = market.get("info", {})
+        if market_info.get("hip3"):
+            configured = self._get_configured_hip3_dexes()
+            if not configured:
+                return False
+
+            market_dex = market_info.get("dex")
+            return market_dex in configured
+
+        return True
+
+    def get_balances(self) -> dict:
+        """Fetch balances from default DEX and HIP-3 DEXes needed by tradable pairs."""
+        balances = super().get_balances()
+
+        for dex in self._get_configured_hip3_dexes():
+            try:
+                dex_balance = self._api.fetch_balance({"dex": dex})
+
+                for currency, amount_info in dex_balance.items():
+                    if currency in ["info", "free", "used", "total", "datetime", "timestamp"]:
+                        continue
+
+                    if currency not in balances:
+                        balances[currency] = amount_info
+                    else:
+                        balances[currency]["free"] += amount_info["free"]
+                        balances[currency]["used"] += amount_info["used"]
+                        balances[currency]["total"] += amount_info["total"]
+
+            except Exception as e:
+                logger.error(f"Could not fetch balance for HIP-3 DEX '{dex}': {e}")
+
+        return balances
```

Comment:
How exactly did you test this? 

as far as i understand - all hip3 pairs are perpetuals.
you'll never have actual "balance" on these - you'll have open positions - so calling this for other dexes is pointless - and in my testing - having or removing this whole method doesn't do much.
the only thing is that the wallets "used" and "total" fields are correct - but they're not used anywhere  (not in futures mode, anyway) - so i'm not sure it's worth keeping this considering it's a rather expensive call.